text/0001-package-structure.md

```diff
@@ -0,0 +1,135 @@
+
+The purpose of this structure is to split the project into smaller, more manageable parts
+and to differentiate between the "library" part of the project which is reusable
+and the "application" which builds ontop of the library + metrics packages + additional services.
+
+An additional benefit is that it will be easier to refactor code if it is all accessible in the same repository.
+
+## Packages
+### `ref-core`
+This package will be a library containing the core functionality of the REF.
+This package will be a dependency of all other packages as it will describe the interfaces that metrics providers must implement. 
+This allows us to keep the core functionality separate from the metrics providers.
+
+### `ref-metrics-*`
+Each metric provider will implement a separate package that implements the interfaces described by the `ref-core` package.
+These will very thin packages that wrap around the existing APIs of the benchmarking packages or use CMEC to execute.
+
+An example package `ref-metrics-example` has been implemented in [CMIP-REF/cmip-ref#5](https://github.com/CMIP-REF/cmip-ref/pull/5).
+See the [code](https://github.com/CMIP-REF/cmip-ref/tree/basic-interface/packages/ref-metrics-example) for more details.
+
+The purpose of these packages are to provide a Python-based hook to execute a metric and return the results.
+This will include (hopefully thin) wrappers around the metric providers existing APIs
+or build the required CMEC configuration files.
+It should require little to no change to the existing codebase benchmarking packages.
+
+The repository will set up a CI pipeline that will run the unit tests for each package.
+This includes fetching some sample data.
+
+These packages will maintain their own dependencies (python and otherwise) and packaged up as a docker image.
+
+The package will be named `ref-metrics-<provider>` where `<provider>` is the name of the provider.
+
+### `ref-api` (TBD)
+This application package will bring together the core and metrics packages to provide a service that
+responds to new data submissions,
+tracks the execution of metrics and the results.
+The package may have indirect dependencies on the metrics packages to avoid conflicting dependencies.
+That adds additional complexity, but will be more flexible in the long run.
+
+This is the service that will be deployed to ESGF alongside any additional services that are required.
+
+Whether this application can be done solely via a CLI-based tool is up for debate.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* More people working in the same repository may lead to more merge conflicts.
+* Larger repository with more moving parts/concepts
+
+
+Another drawback is that it will be harder to manage different package managers.
+The project currently uses [uv](https://docs.astral.sh/uv/) as a package manager as the core and framework
+will not require any complex to install dependencies, 
+but science packages may require more complex dependencies.
```

Comment:
If I understand it correctly, we're not planning to do any science in the REF. The science should happen in the metrics/diagnostics packages. Therefore I would not expect that we need any science dependencies.

Reply:
Correct. But the ref-metric-* packages would pull those science dependencies in. Those metric provider packages probably should live outside of the `cmip-ref` repository like you suggested for the example package. That example package could then become a template.  It would also provide some flexibility for metrics packages to use whatever package installer they need.

I'll update the text to reflect this.